### PR TITLE
feature flag for web applications initial commit

### PR DESCRIPTION
**Pull request by jonocairns**

### CODE REVIEW

File: docs/HOL-Feature_Flag_for_Web_Applications/README.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+![](<media/placeholder.png>)
+
+Also when you start typing a red border appears around the input. This will continue until you supply a valid input similar to the placeholder pattern.
+
+![](<media/text-error.png>)
+
+![](<media/valid-number.png>)
+
+In this lab you have learned how to implement feature flags in a web application. This gives you the ability to seamlessly manage features in your application. You can give users access to beta functionality to give you better test coverage and deliver continuous improvement.
```
Comment (steven2000nz) [inline docs/HOL-Feature_Flag_for_Web_Applications/README.md]: Point to the main website HOL (i.e., Deployment.md) for how to publish it to Azure and test there too.

File: docs/HOL-Feature_Flag_for_Web_Applications/README.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+    }
+
+Now let's explain the sections of the above code. 
+
+- **Features** is a list of features we want to use with the application.
+- **GetStatusByKey** is a method we will use to get the current status for a particular key.
+- **ChangeFeatureToggles** takes in a list of selected feature toggles. If the key exists in any of our 'features' that status will be toggled.
+
+*Note: It's a much better idea to store these flags in a database. For simplicity's sake we have stored them in memory.*
```
Comment (steven2000nz) [inline docs/HOL-Feature_Flag_for_Web_Applications/README.md]: Please mention that turning the flag on and off can be externalized from the app

File: docs/HOL-Feature_Flag_for_Web_Applications/README.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+**Step 8.** Navigate back to the feature flag page (Profile -> View Features). Now toggle the flag to be on. Let's see if our changes are pushed through.
+
+ **Step 9.** Navigate back to the user management page by clicking on this [link](http://localhost:60237/Manage).
+
+Notice the placeholder is different to the one before.
+
+![](<media/placeholder.png>)
+
+Also when you start typing a red border appears around the input. This will continue until you supply a valid input similar to the placeholder pattern.
```
Comment (steven2000nz) [inline docs/HOL-Feature_Flag_for_Web_Applications/README.md]: Can you please add an error message as well - e.g. letters not allowed

File: docs/HOL-Feature_Flag_for_Web_Applications/README.md (hunk 1 above)
Comment (jonocairns) [inline docs/HOL-Feature_Flag_for_Web_Applications/README.md]: Done

File: docs/HOL-Feature_Flag_for_Web_Applications/README.md (hunk 2 above)
Comment (jonocairns) [inline docs/HOL-Feature_Flag_for_Web_Applications/README.md]: Done

File: docs/HOL-Feature_Flag_for_Web_Applications/README.md (hunk 3 above)
Comment (jonocairns) [inline docs/HOL-Feature_Flag_for_Web_Applications/README.md]: Done

**Pull request closed without merging**